addOns/ascanrulesBeta/src/test/java/org/zaproxy/zap/extension/ascanrulesBeta/EnvFileScanRuleUnitTest.java

```diff
@@ -117,4 +136,76 @@ void shouldNotAlertWhenRequestIsSuccessfulButContentNotRelevant() throws Excepti
         // Then
         assertThat(alertsRaised, hasSize(0));
     }
+
+    @ParameterizedTest
+    @EmptySource
+    @ValueSource(strings = {NanoHTTPD.MIME_HTML, NanoHTTPD.MIME_PLAINTEXT})
+    void shouldNotAlertWhenRequestIsSuccessfulButContentIsRelevantWrongType(String input) throws Exception {
+        // Given
+        String path = "/.env";
+        this.nano.addHandler(
+                createHandler(
+                        path,
+                        newFixedLengthResponse(
+                                Response.Status.OK, input, RELEVANT_BODY)));
+        HttpMessage msg = this.getHttpMessage(path);
+        rule.init(msg, this.parent);
+        // When
+        rule.scan();
+        // Then
+        assertThat(alertsRaised, hasSize(0));
+    }
+
+    @Test
+    void shouldNotAlertWhenResponseIsNotOK() throws Exception {
+        // Given
+        String path = "/.env";
+        this.nano.addHandler(
+                createHandler(
+                        path,
+                        newFixedLengthResponse(
+                                Response.Status.BAD_REQUEST,
+                                null,
+                                RELEVANT_BODY)));
+        HttpMessage msg = this.getHttpMessage(path);
+        rule.init(msg, this.parent);
+        // When
+        rule.scan();
+        // Then
+        assertThat(alertsRaised, hasSize(0));
+    }
+
+    @Test
+    void shouldNotAlertWhenResponseIsTooLong() throws Exception {
+        // Given
+        String path = "/.env";
+        this.nano.addHandler(
+                createHandler(
+                        path,
+                        newFixedLengthResponse(
+                                Response.Status.OK, null, IRRELEVANT_BODY + RELEVANT_BODY)));
+        HttpMessage msg = this.getHttpMessage(path);
+        rule.init(msg, this.parent);
+        // When
+        rule.scan();
+        // Then
+        assertThat(alertsRaised, hasSize(0));
+    }
+
+    @Test
+    void shouldNotAlertWhenResponseIsTooLongAndNotOK() throws Exception {
```

Comment:
This one should not be needed.

Reply:
resolved

Reply:
hmm, I swear I updated this, but i just changed again and repushed.